pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -384,6 +405,140 @@ func (r *ReconcileComplianceScan) phaseLaunchingHandler(h scanTypeHandler, logge
 	return reconcile.Result{}, nil
 }
 
+// getRuntimeKubeletConfig gets the runtime kubeletconfig for a node
+// by fetching /api/v1/nodes/$nodeName/proxy/configz endpoint use the kubeClientset
+// and store it in a configmap for each node
+func (r *ReconcileComplianceScan) getRuntimeKubeletConfig(nodeName string) (string, error) {
+	// get the runtime kubeletconfig
+	kubeletConfigIO, err := r.ClientSet.CoreV1().RESTClient().Get().RequestURI("/api/v1/nodes/" + nodeName + "/proxy/configz").Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer kubeletConfigIO.Close()
+	kubeletConfig, err := ioutil.ReadAll(kubeletConfigIO)
+	if err != nil {
+		return "", err
+	}
+	// kubeletConfig is a byte array, we need to convert it to string
+	kubeletConfigStr := string(kubeletConfig)
+	return kubeletConfigStr, nil
+}
+
+func (r *ReconcileComplianceScan) createKubeletConfigCM(instance *compv1alpha1.ComplianceScan, node *corev1.Node) error {
+	kubeletConfig, err := r.getRuntimeKubeletConfig(node.Name)
+	if err != nil {
+		return fmt.Errorf("cannot get the runtime kubelet config for node %s: %v", node.Name, err)
+	}
+	// create a configmap for the node
+	kubeletConfigCM := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getKubeletCMNameForScan(instance, node),
+			Namespace: instance.Namespace,
+			CreationTimestamp: metav1.Time{
+				Time: time.Now(),
+			},
```

Comment:
We could make this Config Map immutable

https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-immutable